.devcontainer/.env.codespaces

```diff
@@ -0,0 +1,14 @@
+# This is the .env that is used by the codespaces environment.
+# See the .devcontainer/devcontainer.json file for how this is configured.
+PG_USERNAME=postgres
+PG_PASSWORD=postgres
+DELAYED_JOB_USERNAME='admin'
+DELAYED_JOB_PASSWORD='password'
+FLIPPER_USERNAME="admin"
+FLIPPER_PASSWORD="password"
+RECAPTCHA_SITE_KEY=6LeIxAcTAAAAAJcZVRqyHh71UMIEGNQ_MXjiZKhI
+RECAPTCHA_PRIVATE_KEY=6LeIxAcTAAAAAGG-vFI1TnRWxMZNFuojJ4WifJWe
+
+# [OPTIONAL] - Used to fetch copies of production DB
+AZURE_STORAGE_ACCOUNT_NAME=
+AZURE_STORAGE_ACCESS_KEY=
```

Comment:
Not familiar with codespaces... how does this interact with the `.env` file in the base directory? Does that override this one?

Reply:
When the codespace starts up, it calls the [post-create.sh](https://github.com/rubyforgood/human-essentials/pull/4292/files#diff-bcf66a848607b49d2a45edfad434f6808017ee9d18f5a20dde6670745010d89aR3) script. There is a line that copies this file to `.env`.

So yes, it creates the one in the base directory and ensures it is compatible with this codespace.

Now that I think about it, I should probably do `cp -f` so that the command doesn't fail if `.env` already exists (even though it shouldn't).

Reply:
Actually, I'd want to make sure that the local `.env` (which is gitignored) overrides the one that's checked in.

Reply:
Even though there's no realistic scenario that I can think of where this would happen, I can make that change. This script is only run when the codespace is created. In this case, there would be no `.env` file yet.

Reply:
Actually, now that I think about it, there is a scenario. If someone runs the `Codespace: Rebuild` command, then the script will be rerun on the existing file system. This command is only really used when you're making changes to the codespaces config.

Reply:
The point is that local `.env` might contain values that we want to override whatever is specified in the codespaces `.env`. For example, I have Azure logins on my local which I don't want to check into git and if I _did_ use codespaces, I'd want them to override the defaults that are checked in. Does that make sense?

Reply:
I think this is the scenario that you are thinking of. Please tell me if I understand correctly:

1. You have a local checkout of `human-essentials` with a custom `.env` that contains some azure credentials
2. You create a codespace and the start up scripts generate a `.env` for this codespace.
3. Now you want to use the credentials in your local `.env` in your codespace `.env`.

In this case, I think you'd need to manually copy from your local to your codespace. But I don't think this is the most common use case. The use case for the codespace support (as I see it) is to give newcomers a one-click way to get a dev environment. If you already have a local dev environment that works, is there a reason to use a codespace as well?

Reply:
The point is more that `.env` can contain things that aren't just to get things working. They can also override defaults, so different people might have different values in there. In my case, I might not have a working local copy, but at some point I'm going to want to be able to access Azure, and I'd have no way of doing it via codespaces, it'd mean I'd have to abandon it and go down the local route.

Again - never used codespaces, not sure how important this is. Still waiting for @awwaiid to weigh in on this.

Reply:
Thanks. I understand your concern. 

The script to create the `.env` file only runs when the codespace is created. At this point, we are guaranteed that `.env` does not yet exist (unless someone accidentally commits one). Once the `.env` in the codespace is created, you can customize it in any way you like. The script is never re-run (except if a user explicitly calls the vscode command to rebuild the codespace). Now that I think about it, this is a good reason to make sure that `.env` is not overwritten if it already exists.